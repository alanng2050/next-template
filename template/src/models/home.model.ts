import { createModel } from '@rematch/core'
import { RootModel } from '.'

interface IData {
  user: User
}
const defaultValues: IData = {
  user: {
    name: 'Bee',
    avatar: '/avatar.png',
  },
}

export const home = createModel<RootModel>()({
  state: defaultValues,
  reducers: {
    updateName(s, payload: User) {
      s.user = payload
    },
  },
  effects: (dispatch) => ({}),
})
