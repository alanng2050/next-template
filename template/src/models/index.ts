import immerPlugin from '@rematch/immer'
import { init, Models, RematchDispatch, RematchRootState } from '@rematch/core'
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux'
import { home } from './home.model'

export interface RootModel extends Models<RootModel> {
  home: typeof home
}

export const models: RootModel = { home }

export const store = init<RootModel>({
  models,
  // add immerPlugin to your store
  plugins: [immerPlugin()],
})

export type Store = typeof store
export type Dispatch = RematchDispatch<RootModel>
export type RootState = RematchRootState<RootModel>

export const useStoreDispatch = () => useDispatch<Dispatch>()
export const useStoreState: TypedUseSelectorHook<RootState> = useSelector
