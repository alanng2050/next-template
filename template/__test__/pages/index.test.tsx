import { render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import Home from '@/pages/index'
import { store } from '@/models'

describe('Home', () => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }

  it('should render correctly', () => {
    const { asFragment } = render(<Home />, { wrapper: Wrapper })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should see the text Bee', () => {
    render(<Home />, { wrapper: Wrapper })
    expect(screen.getByText('This is Bee')).toBeInTheDocument()
  })
})
