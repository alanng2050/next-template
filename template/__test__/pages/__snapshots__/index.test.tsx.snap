// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gsqmhk"
  >
    <div
      class="user_card css-gmuwbf"
    >
      <div
        class="chakra-card css-2dlz7"
      >
        <div
          class="chakra-card__body css-0"
        >
          <p
            class="chakra-text css-1mzsble"
          >
            This is Bee
          </p>
          <img
            alt="avatar"
            class="chakra-image css-v7v99c"
            src="/avatar.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
